KMD.Identity.TestApplications.OpenID.Angular/ClientApp/src/app/app.component.html

```diff
@@ -23,37 +23,28 @@
   </nav>
 </header>
 
-
 <div role="main">
   <div>
-    <div *ngIf="error">
-      <app-error [error]="error"></app-error>
-    </div>
-    <div *ngIf="isAuthenticated; else authenticationPrompt">
-      <div *ngIf="performUserDelegation">
-        <app-user-delegation></app-user-delegation>
-      </div>
-      <div *ngIf="showTestApiCall">
-        <app-test-api-call></app-test-api-call>
-      </div>
-      <app-user-claims [userClaims]="userData"></app-user-claims>
+    <div *ngIf="isError">
+      <app-error></app-error>
     </div>
+      <ng-container *ngIf="requireAuthentication(); then authenticationPrompt; else authenticated"></ng-container>
```

Comment:
Note here, `ng-container` allows a simpler approach to "if this then show this else show this"

`authenticationPrompt` is an `ng-template`, as is `authenticated`. The `ng-container` is essentially a wrapper that can work with templates